### PR TITLE
Add behaviour

### PR DESCRIPTION
**Pull request by schas002**
Add behaviour to the chatbot:

- When a message contains `hello`, reply with `Hello!`
- Add three commands: `!!zhelp`, `!!zlistcommands`, `!!zalive`.

### CODE REVIEW

File: chatbot.js [hunk 1]
```diff
@@ -80,6 +80,18 @@ var Chatbot = function Chatbot() {
...
+var DICTIONARY = { // Define commands in this object
+	"!!zhelp": "No help yet, but you can use !!zlistcommands.",
+	"!!zlistcommands": "Commands: listcommands, alive",
+	"!!zalive": "Watching this endless feed of new messages never gets boring."
+};
+
 var MyChatbot = new Chatbot("a Chatbot", {Startup: "Hi my name is $Name!"}, function() {
-	
+	if(/hello/i.test(this.Text)) { this.Reply("Hello!"); }
+	else if(this.Text.slice(0, 3) !== "!!z") { /* Nothing to do here */ }
```
Comment (vihanb) [inline chatbot.js]: can't this become:

```
else if(this.Text.slice(0, 3) === "!!z") {
                var text = DICTIONARY[this.Text];
		if(text) { this.Reply(text); }
		else { this.Reply("Unknown command"); }
}
```

Comment (schas002) [inline chatbot.js]: Good point :)

File: chatbot.js [hunk 2]
```diff
@@ -80,6 +80,18 @@ var Chatbot = function Chatbot() {
...
 
+var DICTIONARY = { // Define commands in this object
+	"!!zhelp": "No help yet, but you can use !!zlistcommands.",
+	"!!zlistcommands": "Commands: listcommands, alive",
+	"!!zalive": "Watching this endless feed of new messages never gets boring."
+};
+
 var MyChatbot = new Chatbot("a Chatbot", {Startup: "Hi my name is $Name!"}, function() {
-	
+	if(/hello/i.test(this.Text)) { this.Reply("Hello!"); }
```
Comment (vihanb) [inline chatbot.js]: This could even become `this.Reply("Hello " + this.User +"!")` so you can get their name.

File: chatbot.js [hunk 3]
```diff
@@ -80,6 +80,18 @@ var Chatbot = function Chatbot() {
   }, 2500);
 };
 
+var DICTIONARY = { // Define commands in this object
```
Comment (vihanb) [inline chatbot.js]: perhaps make this a `Map` object which has nicer comments like `.has` `.get` `.set` and `.forEach` to loop through it.

```js
new Map([
        ["!!zhelp", "No help yet, but you can use !!zlistcommands."],
 	["!!zlistcommands", "Commands: listcommands, alive"],
 	["!!zalive", "Watching this endless feed of new messages never gets boring."]
])
```

File: chatbot.js [hunk 4]
```diff
@@ -80,6 +80,18 @@ var Chatbot = function Chatbot() {
...
+	"!!zlistcommands": "Commands: listcommands, alive",
+	"!!zalive": "Watching this endless feed of new messages never gets boring."
+};
+
 var MyChatbot = new Chatbot("a Chatbot", {Startup: "Hi my name is $Name!"}, function() {
-	
+	if(/hello/i.test(this.Text)) { this.Reply("Hello!"); }
+	else if(this.Text.slice(0, 3) !== "!!z") { /* Nothing to do here */ }
+	else {
+		var text = DICTIONARY[this.Text];
```
Comment (vihanb) [inline chatbot.js]: you can golf this too:

```js
this.Reply(
    DICTIONARY[this.Text] || "Unknown  Command"
)
```

or if you choose to make it a `Map`:

```js
this.Reply(
    DICTIONARY.get(this.Text) || "Unknown  Command"
)
```

**Comment by vihanb** [comment]
Command proposal: `!!zgoat` which prints picture of a goat from google images ;)

**Comment by schas002** [comment]
Maybe not? Maybe put it in a separate pull request?

File: chatbot.js (hunk 3 above)
Comment (schas002) [inline chatbot.js]: How should I loop through the `Map`?

**Comment by schas002** [comment]
Applied three of four of your proposed refactors, @vihanb. Time to do this.

**Pull request merged**